src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -770,6 +772,26 @@ private EntityHandle GetMemberReferenceHandle(MemberInfo memberInfo)
             }
 
             return memberHandle;
+
+            // The field type has to be the original open generic field type, not the closed type.
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern", Justification = "Internal reflection implementation")]
+            static Type GetFieldType(FieldInfo fieldInfo)
+            {
+                if (fieldInfo.DeclaringType!.IsGenericType)
```

Comment:
```suggestion
                if (fieldInfo.DeclaringType!.IsConstructedGenericType)
```

Reply:
With the switch to using GetOriginalMemberIfConstructedType(), this is no longer needed.